addons/macrochanger/macrochanger.lua

```diff
@@ -26,143 +26,85 @@
-            BST = {Book = '9', Page = '1'},
-            BRD = {Book = '10', Page = '1'},
-            RNG = {Book = '11', Page = '1'},
-            SAM = {Book = '12', Page = '1'},
-            NIN = {Book = '13', Page = '1'},
-            DRG = {Book = '14', Page = '1'},
-            SMN = {Book = '15', Page = '1'},
-            BLU = {Book = '16', Page = '1'},
-            COR = {Book = '17', Page = '1'},
-            PUP = {Book = '18', Page = '1'},
-            DNC = {Book = '19', Page = '1'},
-            SCH = {Book = '20', Page = '1'},
-            GEO = {Book = '20', Page = '1'},
-            RUN = {Book = '20', Page = '1'},
-            }
-		print('Default settings file created')
-		windower.add_to_chat(12,'MacroChanger created a settings file and loaded!')
-	else
-		f:close()
-		for curline in io.lines(windower.addon_path..'data/settings.txt') do
-			local splat = curline:gsub(':',''):split(' ')
-			local cmd = ''
-			if splat[1] and macros[splat[1]:upper()] and splat[2] ~=nil and (splat[2]:lower() == 'book' or splat[2]:lower() == 'page') and splat[3] then
-				macros[splat[1]:upper()][splat[2]:ucfirst()] = splat[3] -- Instead of a number, this can also be 'disabled'
-			elseif splat[1] and splat[2] and (splat[1]..' '..splat[2]) == 'disable all' and tonumber(splat[3]) then
-				globaldisable = tonumber(splat[3])
-			end
-		end
-		windower.add_to_chat(12,'MacroChanger read from a settings file and loaded!')
-	end
+function addon_command (...)
+  local args = {...}
+  if args[1]:lower() == 'disableall' then
+    if args[2]:lower() == 'on' then
+      settings.globaldisable = 1
+      windower.add_to_chat(17, 'All automated macro switching disabled.')
+    elseif args[2]:lower() == 'off' then
+      settings.globaldisable = 0
+      windower.add_to_chat(17, 'Automated macro switching enabled.')
+    end
+   elseif args[1]:lower() == 'help' then
+     windower.add_to_chat(17, 'MacroChanger Commands:')
+     windower.add_to_chat(17, 'disableall [on|off]')
+     windower.add_to_chat(17, '   on - Disables all automated macro switching')
+     windower.add_to_chat(17, '   off - Enables all automated macro switching not disabled individually')
+     windower.add_to_chat(17, '   Resets to what is stored in settings upon unloading of addon.  To Permanently change, please change the option in the settings file.')
+   end
 end
 
-windower.register_event('job change',function ()
+function job_change ()
 -- Could use the job ID passed into this function, but the addon would have to include the resources library
-	local job = windower.ffxi.get_player().main_job
-	local book = ''
-	local page = ''
-	if globaldisable == 0 then
-        if job and macros[job] then
-			book = macros[job].Book
-			page = macros[job].Page
-		end
+  local mjob = windower.ffxi.get_player().main_job
+  local sjob = windower.ffxi.get_player().sub_job
+  local job  = ''
+  local book = ''
+  local page = ''
+  if settings.globaldisable == 0 then
```

Comment:
This looks like a boolean setting, so you can use a boolean variable. Will also make it easier on users, since this is written to their user files, so they can use `true` and `false` instead of `1` and `0`.